app/code/Magento/MediaContentCatalog/Model/GetAssetIdByEavContentField.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MediaContentCatalog\Model;
+
+use Magento\Eav\Model\Config;
+use Magento\Framework\App\ResourceConnection;
+use Magento\MediaContentApi\Model\GetAssetIdByContentFieldInterface;
+
+/**
+ * Class responsible to return Asset id by eav content field
+ */
+class GetAssetIdByEavContentField implements GetAssetIdByContentFieldInterface
+{
+    private const TABLE_CONTENT_ASSET = 'media_content_asset';
+
+    /**
+     * @var ResourceConnection
+     */
+    private $connection;
+
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var string
+     */
+    private $attributeCode;
+
+    /**
+     * @var string
+     */
+    private $entityType;
+
+    /**
+     * @var array
+     */
+    private $valueMap;
+
+    /**
+     * GetAssetIdByEavContentStatus constructor.
+     *
+     * @param ResourceConnection $resource
+     * @param Config $config
+     * @param string $attributeCode
+     * @param string $entityType
+     * @param array $valueMap
+     */
+    public function __construct(
+        ResourceConnection $resource,
+        Config $config,
+        string $attributeCode,
+        string $entityType,
+        array $valueMap = []
+    ) {
+        $this->connection = $resource;
+        $this->config = $config;
+        $this->attributeCode = $attributeCode;
+        $this->entityType = $entityType;
+        $this->valueMap = $valueMap;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute(string $value): array
+    {
+        $statusAttribute = $this->config->getAttribute($this->entityType, $this->attributeCode);
```

Comment:
That may be not a status attribute
```suggestion
        $attribute = $this->config->getAttribute($this->entityType, $this->attributeCode);
```